### PR TITLE
Fix the invalid read access for fully ionized ions in the ADK implementation

### PR DESCRIPTION
**Pull request by n01r**
When the ion is fully ionized the trait `GetIonizationEnergies` will not
have a valid target for read access in the vector of ionization energies.
An `if`-condition avoid that access has been added to the ionization model ADK.
(*This change happens according to @psychocoderHPC 's [comment](https://github.com/ComputationalRadiationPhysics/picongpu/pull/1423#discussion-diff-64031848R72)*)

Also equals sign alignment has been removed as it is not part of our
coding style guideline.

### CODE REVIEW

**Comment by PrometheusPi** [comment]
There is a typo `reas -> read` in your commit message.

File: src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp [hunk 1]
```diff
@@ -63,44 +63,49 @@ namespace ionization
...
+                float_X eInAU = math::abs(eField) / ATOMIC_UNIT_EFIELD;
 
-            /* ionization rate */
-            float_X rateADK     = math::sqrt(float_X(3.0) * util::cube(nEff) * eInAU / (pi * util::cube(protonNumber))) \
-                                    * eInAU * util::square(dFromADK) / (float_X(8.0) * pi * protonNumber) \
-                                    * math::exp(float_X(-2.0) * util::cube(protonNumber) / (float_X(3.0) * util::cube(nEff) * eInAU));
+                /* effective principal quantum number (unitless) */
+                float_X nEff = protonNumber / math::sqrt(float_X(2.0) * iEnergy );
+                /* nameless variable for convenience dFromADK*/
+                float_X dBase = float_X(4.0) * util::cube(protonNumber) / ( eInAU * util::quad(nEff)) ;
```
Comment (PrometheusPi) [inline src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp]: please either remove the space after `(` or add a space behind `)`
standard:
```c++
( ... )
#or
(...)
```

**Comment by PrometheusPi** [comment]
looks good to me - please add minor changes :+1:

**Comment by n01r** [comment]
I took the liberty to just do `git commit --amend`

**Pull request merged**